api/rest/restapi.go

```diff
@@ -497,24 +497,35 @@ func (api *API) graphHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) addFileHandler(w http.ResponseWriter, r *http.Request) {
-	contentType := r.Header.Get("Content-Type")
-	mediatype, _, _ := mime.ParseMediaType(contentType)
-
-	if mediatype != "multipart/form-data" {
-		sendErrorResponse(w, 415, "unsupported media type")
-		return
-	}
 	reader, err := r.MultipartReader()
 	if err != nil {
 		sendErrorResponse(w, 400, err.Error())
 		return
 	}
 
-	params := r.URL.Query()
+	urlParams := r.URL.Query()
+	layout := urlParams.Get("layout")
+	chunker := urlParams.Get("chunker")
+	raw, _ := strconv.ParseBool(urlParams.Get("raw"))
+	hidden, _ := strconv.ParseBool(urlParams.Get("hidden"))
+	shard, _ := strconv.ParseBool(urlParams.Get("shard"))
+	replMin, _ := strconv.Atoi(urlParams.Get("repl_min"))
+	replMax, _ := strconv.Atoi(urlParams.Get("repl_max"))
```

Comment:
It might be worth doing error checking for these params, making sure they could be parsed and that they have allowable values.